apps/performance/memoized/src/app/fib.pipe.ts

```diff
@@ -0,0 +1,28 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+const fibonacci = (num: number): number => {
+  if (num === 1 || num === 2) {
+    return 1;
+  }
+  return fibonacci(num - 1) + fibonacci(num - 2);
+};
+@Pipe({
+  standalone: true,
+  name: 'Fibonacci',
+})
+export class FibonacciPipe implements PipeTransform {
+  /**
+   *
+   * @param {}
+   * @param fibObject
+   * Cant create fibObject here because on every pipe call it is creating new object instance so creating from person-list
+   * and sending from there.
+   */
+  transform(value: number, fibObject: any) {
+    if (fibObject[value]) {
+      return fibObject[value];
+    }
+    fibObject[value] = fibonacci(value);
```

Comment:
What is the goal of this fibObject ?

Reply:
I did not wanted to calculate Fibonacci of same numbers so stored them in object to reuse so that if number appears again it will get from object and will not calculate.